code/datums/spells/spacetime_dist.dm

```diff
@@ -0,0 +1,160 @@
+// This could probably be an aoe spell but it's a little cursed, so I'm not touching it
+/obj/effect/proc_holder/spell/spacetime_dist
+	name = "Spacetime Distortion"
+	desc = "Entangle the strings of space-time in an area around you, \
+		randomizing the layout and making proper movement impossible. The strings vibrate... \
+		Upgrading the spell increases range, it does not lower cooldown."
+	sound = 'sound/magic/strings.ogg'
+	action_icon_state = "spacetime"
+
+	school = "transmutation"
+	base_cooldown = 30 SECONDS
+	clothes_req = TRUE
+	invocation = "none"
+	centcom_cancast = FALSE //Prevent people from getting to centcom
+	cooldown_min = 30 SECONDS //No reduction, just more range.
+	level_max = 3
+
+	/// Whether we're ready to cast again yet or not. In the event someone lowers their cooldown with charge.
+	var/ready = TRUE
+	/// The radius of the scramble around the caster. Increased by 3 * spell_level
+	var/scramble_radius = 7
+	/// The duration of the scramble
+	var/duration = 15 SECONDS
+	/// A lazylist of all scramble effects this spell has created.
+	var/list/effects
+
+/obj/effect/proc_holder/spell/spacetime_dist/Destroy()
+	QDEL_LIST(effects)
+	return ..()
+
+/obj/effect/proc_holder/spell/spacetime_dist/create_new_targeting()
+	return new /datum/spell_targeting/self
+
+/obj/effect/proc_holder/spell/spacetime_dist/cast_check(charge_check = TRUE, start_recharge = TRUE, mob/user = usr)
+	return ..() && ready
+
+/obj/effect/proc_holder/spell/spacetime_dist/cast(list/targets, mob/user = usr)
+	. = ..()
+	var/list/turf/to_switcharoo = get_targets_to_scramble(usr)
+	if(!length(to_switcharoo))
+		to_chat(usr, "<span class='warning'>For whatever reason, the strings nearby aren't keen on being tangled.</span>")
+		return
+
+	ready = FALSE
+
```

Comment:
```suggestion

	effects = list()
```
Then you can ditch the `LAZYADD` calls and just do `effects += effect_a` and for b.
I don't think this list is really lazy either due to the fact that it gets created when the effect starts. Then just gets unset again when it deactivates

Reply:
done